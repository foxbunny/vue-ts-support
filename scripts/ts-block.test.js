const ts = require('./ts-block.js')
const mockContext = require('./test-helpers').mockContext


test('default options', () => {
  expect(typeof ts({})).toBe('function')
})

;[1, 2].forEach((webpackVer) => {
  test(`default options on webpack ${webpackVer}`, () => {
    const blockfn = ts({})
    expect(blockfn(mockContext(webpackVer))).toMatchSnapshot()
  })

  test(`arbitrary options on webpack ${webpackVer}`, () => {
    const blockfn = ts({ appendTsSuffixTo: [/\.vue$/] })
    expect(blockfn(mockContext(webpackVer))).toMatchSnapshot()
  })
})
