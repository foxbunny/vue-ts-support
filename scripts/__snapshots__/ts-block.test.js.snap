// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`arbitrary options on webpack 1 1`] = `
Object {
  "module": Object {
    "loaders": Array [
      Object {
        "loader": "ts-loader",
        "options": Object {
          "appendTsSuffixTo": Array [
            /\\\\\\.vue\\$/,
          ],
        },
        "test": undefined,
      },
    ],
  },
  "resolve": Object {
    "extensions": Array [
      ".ts",
    ],
  },
}
`;

exports[`arbitrary options on webpack 2 1`] = `
Object {
  "module": Object {
    "loaders": Array [
      Object {
        "loader": "ts-loader",
        "options": Object {
          "appendTsSuffixTo": Array [
            /\\\\\\.vue\\$/,
          ],
        },
        "test": undefined,
      },
    ],
  },
  "resolve": Object {
    "extensions": Array [
      ".ts",
    ],
  },
}
`;

exports[`default options on webpack 1 1`] = `
Object {
  "module": Object {
    "loaders": Array [
      Object {
        "loader": "ts-loader",
        "options": Object {},
        "test": undefined,
      },
    ],
  },
  "resolve": Object {
    "extensions": Array [
      ".ts",
    ],
  },
}
`;

exports[`default options on webpack 2 1`] = `
Object {
  "module": Object {
    "loaders": Array [
      Object {
        "loader": "ts-loader",
        "options": Object {},
        "test": undefined,
      },
    ],
  },
  "resolve": Object {
    "extensions": Array [
      ".ts",
    ],
  },
}
`;
